llama_index/embeddings/dashscope.py

```diff
@@ -0,0 +1,331 @@
+    model: str, url: str, api_key: Optional[str] = None, **kwargs: Any
+) -> Optional[str]:
+    """Call DashScope batch text embedding.
+
+    Args:
+        model (str): The `DashScopeMultiModalEmbeddingModels`
+        url (str): The url of the file to embedding which with lines of text to embedding.
+
+    Raises:
+        ImportError: Need install dashscope package.
+
+    Returns:
+        str: The url of the embedding result, format ref:
+        https://help.aliyun.com/zh/dashscope/developer-reference/text-embedding-async-api-details
+    """
+    try:
+        import dashscope
+    except ImportError:
+        raise ImportError("DashScope requires `pip install dashscope")
+    response = dashscope.BatchTextEmbedding.call(
+        model=model, url=url, api_key=api_key, kwargs=kwargs
+    )
+    if response.status_code == HTTPStatus.OK:
+        return response.output["url"]
+    else:
+        logger.error("Calling BatchTextEmbedding failed, details: %s" % response)
+        return None
+
+
+def get_multimodal_embedding(
+    model: str, input: list, api_key: Optional[str] = None, **kwargs: Any
+) -> List[float]:
+    """Call DashScope multimodal embedding.
+       ref: https://help.aliyun.com/zh/dashscope/developer-reference/one-peace-multimodal-embedding-api-details.
+
+    Args:
+        model (str): The `DashScopeBatchTextEmbeddingModels`
+        input (str): The input of the embedding, eg:
+             [{'factor': 1, 'text': '你好'},
+             {'factor': 2, 'audio': 'https://dashscope.oss-cn-beijing.aliyuncs.com/audios/cow.flac'},
+             {'factor': 3, 'image': 'https://dashscope.oss-cn-beijing.aliyuncs.com/images/256_1.png'}]
+
+    Raises:
+        ImportError: Need install dashscope package.
+
+    Returns:
+        str: The url of the embedding result, format ref:
+        https://help.aliyun.com/zh/dashscope/developer-reference/text-embedding-async-api-details
+    """
+    try:
+        import dashscope
+    except ImportError:
+        raise ImportError("DashScope requires `pip install dashscope")
+    response = dashscope.MultiModalEmbedding.call(
+        model=model, input=input, api_key=api_key, kwargs=kwargs
+    )
+    if response.status_code == HTTPStatus.OK:
+        return response.output["embedding"]
+    else:
+        logger.error("Calling MultiModalEmbedding failed, details: %s" % response)
+        return []
+
+
+class DashScopeTextEmbedding(BaseEmbedding):
+    """DashScope class for text embedding.
+
+    Args:
+        model (str): Model for embedding.
+            Defaults to DashScopeTextEmbeddingModels.TEXT_EMBEDDING_V2.
+                Options are:
+
+                - DashScopeTextEmbeddingModels.TEXT_EMBEDDING_V1
+                - DashScopeTextEmbeddingModels.TEXT_EMBEDDING_V2
+        api_key (str): The DashScope api key.
+    """
+
+    _api_key: Optional[str] = PrivateAttr()
+
+    def __init__(
+        self,
+        model: str = DashScopeTextEmbeddingModels.TEXT_EMBEDDING_V2,
```

Comment:
nit: can we make this model_name to be consistent with embeddings?

Reply:
updated